bezier-rs/lib/src/lib.rs

```diff
@@ -453,15 +460,89 @@ impl Bezier {
 			.try_into()
 			.unwrap()
 	}
+
+	/// Returns a Bezier curve that results from applying the tranformation function to each point in the Bezier.
+	pub fn apply_transformation(&self, transformation_function: &dyn Fn(DVec2) -> DVec2) -> Bezier {
+		let transformed_start = transformation_function(self.start);
+		let transformed_end = transformation_function(self.end);
+		match self.handles {
+			BezierHandles::Quadratic { handle } => {
+				let transformed_handle = transformation_function(handle);
+				Bezier::from_quadratic_dvec2(transformed_start, transformed_handle, transformed_end)
+			}
+			BezierHandles::Cubic { handle_start, handle_end } => {
+				let transformed_handle_start = transformation_function(handle_start);
+				let transformed_handle_end = transformation_function(handle_end);
+				Bezier::from_cubic_dvec2(transformed_start, transformed_handle_start, transformed_handle_end, transformed_end)
+			}
+		}
+	}
+
+	/// Returns a Bezier curve that results from rotating the curve by the given angle (in radians).
+	pub fn rotate(&self, angle: f64) -> Bezier {
+		let rotation_matrix = DMat2::from_angle(angle);
+		self.apply_transformation(&|point| rotation_matrix.mul_vec2(point))
+	}
+
+	/// Returns a Bezier curve that results from translating the curve by the given `DVec2`.
+	pub fn translate(&self, translation: DVec2) -> Bezier {
+		self.apply_transformation(&|point| point + translation)
+	}
+
+	/// Returns a list of points where the provided `line` intersects with the Bezier curve.
+	/// - `line` - Expected to be received in the format of `[start_point, end_point]`.
+	pub fn line_intersection(&self, line: [DVec2; 2]) -> Vec<DVec2> {
+		// Rotate the bezier and the line by the angle that the line makes with the x axis
+		let slope = line[1] - line[0];
+		let angle = slope.angle_between(DVec2::new(1., 0.));
+		let rotation_matrix = DMat2::from_angle(angle);
+		let rotated_bezier = self.apply_transformation(&|point| rotation_matrix.mul_vec2(point));
+		let rotated_line = [rotation_matrix.mul_vec2(line[0]), rotation_matrix.mul_vec2(line[1])];
+
+		// Translate the bezier such that the line becomes aligned on top of the x-axis
+		let vertical_distance = rotated_line[0].y;
+		let translated_bezier = rotated_bezier.translate(DVec2::new(0., -vertical_distance));
+
+		// Compute the roots of the resulting bezier curve
+		let list_intersection_t = match translated_bezier.handles {
+			BezierHandles::Quadratic { handle } => {
+				let a = translated_bezier.start.y - 2. * handle.y + translated_bezier.end.y;
+				let b = 2. * (handle.y - translated_bezier.start.y);
+				let c = translated_bezier.start.y;
+				let discriminant = b * b - 4. * a * c;
+				let two_times_a = 2. * a;
+				utils::solve_quadratic(discriminant, two_times_a, b, c)
```

Comment:
Just a style suggestion in general, please try and break logical groups of lines into their own sections with blank lines. For example the `let a/b/c` lines, `let discriminant/two_times_a`, and `utils::solve_quadratic` lines seem like good groups here.

An exception to this suggestion is above though, at the top of this function, where there's a comment referring to the `let slope` through `let rotated_line` lines. Even though it would be nice to break the `let slope/angle` and `let rotation_*` lines, the need to keep that comment together with its group is of more importance. I'm just mentioning this to provide some more context of how I like to think of the tradeoffs of when and when to not use blank lines to separate things. Use your discretion but let's also try and keep the code breathable instead of being too compact. So far I've seen a pretty good job being done already in this regard, so I'm just bringing it up because I noticed here where it could be done.